In [2]:
%load_ext sql

In [3]:
%sql mysql+pymysql://root:DSEfe0507727182@127.0.0.1:3306/uni_admission

In [8]:
%%sql 

SELECT *
FROM university_admission_russia
LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
5 rows affected.


[(1, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Mathematics and Mechanics', 2014, 47923, 7144),
 (2, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Computer Science', 2014, 28787, 3238),
 (3, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Physics and Astronomy', 2014, 23906, 4391),
 (4, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Chemistry', 2014, 25930, 2367),
 (5, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Earth Science', 2014, 43434, 6745)]

In [6]:
%%sql 

SELECT DISTINCT Year
FROM university_admission_russia

Environment variable $DATABASE_URL not set, and no connect string given.
Connection info needed in SQLAlchemy format, example:
               postgresql://username:password@hostname/dbname
               or an existing connection: dict_keys([])


In [21]:
%%sql 

SELECT DISTINCT `Number of Applications`
FROM university_admission_russia
LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
5 rows affected.


[(47923,), (28787,), (23906,), (25930,), (43434,)]

In [9]:
%%sql 

SELECT SUM(`Number of Applications`)
FROM university_admission_russia

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
1 rows affected.


[(Decimal('81242082'),)]

The above is the gross sum of all the application in the Uni.

In [11]:
%%sql 

SELECT SUM(`Number of Applications`) as Total_No_Application
FROM university_admission_russia

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
1 rows affected.


[(Decimal('81242082'),)]

The above is the gross sum of all the application in the Uni. N.B: That adding the Alliance syntax "as Total_No_Application" makes the column title different.

### The total applications per year in the data set. 

In [16]:
%%sql 

SELECT Year, SUM(`Number of Applications`) as Total_No_Application
FROM university_admission_russia
GROUP BY Year

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, Decimal('5909095')),
 (2015, Decimal('5946155')),
 (2016, Decimal('6591943')),
 (2017, Decimal('6734044')),
 (2018, Decimal('7313275')),
 (2019, Decimal('7689487')),
 (2020, Decimal('9004885')),
 (2021, Decimal('9699511')),
 (2022, Decimal('10682268')),
 (2023, Decimal('11671419'))]

In [20]:
%%sql 

SELECT Year, SUM(`Number of Applications`) as Total_No_Application
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application DESC

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2023, Decimal('11671419')),
 (2022, Decimal('10682268')),
 (2021, Decimal('9699511')),
 (2020, Decimal('9004885')),
 (2019, Decimal('7689487')),
 (2018, Decimal('7313275')),
 (2017, Decimal('6734044')),
 (2016, Decimal('6591943')),
 (2015, Decimal('5946155')),
 (2014, Decimal('5909095'))]

The above is the total number of application per year in the Uni admission. The table is arranged with the highest to the lowest (DESC ORDER). The highest number of applications were received in 2023 and the lowest received in 2014.

In [22]:
%%sql 

SELECT `Branches of Science` as Faculty, SUM(`Number of Applications`) as Total_No_Application
FROM university_admission_russia
GROUP BY Faculty
ORDER BY Total_No_Application DESC

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[('Engineering and Technology', Decimal('30163032')),
 ('Social Sciences', Decimal('24557142')),
 ('Healthcare and Medicine', Decimal('7357729')),
 ('Education Sciences and Pedagogy', Decimal('7112929')),
 ('Humanities', Decimal('3523491')),
 ('Mathematical and Natural Sciences', Decimal('3501681')),
 ('Agriculture', Decimal('2512689')),
 ('Arts and Culture', Decimal('2410148')),
 ('Oriental and African Studies', Decimal('102459')),
 ('Military Science', Decimal('782'))]

The ten years (2014-2023) total number of applications in the Uni per Faculty (Branches of Science) arranged in a Desc ORDER. The analysis revealed that Engineering and Technology had the highest number of applications while Military Science had the least. 

In [23]:
%%sql 

SELECT DISTINCT `Tuition fees`
FROM university_admission_russia

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
2 rows affected.


[('Scholarship / tuition-free',), ('Full pay / tuition fee',)]

In [24]:
%%sql 

SELECT COUNT(`Tuition fees`)
FROM university_admission_russia

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
1 rows affected.


[(5016,)]

In [25]:
%%sql 

SELECT DISTINCT `Tuition fees`
FROM university_admission_russia

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
(pymysql.err.ProgrammingError) (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DISTINCT `Tuition fees`\nFROM university_admission_russia' at line 1")
[SQL: SELECT COUNT DISTINCT `Tuition fees`
FROM university_admission_russia]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [26]:
%%sql 

SELECT `Tuition fees`, COUNT(`Tuition fees`)
FROM university_admission_russia
GROUP BY `Tuition fees`

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
2 rows affected.


[('Scholarship / tuition-free', 2736), ('Full pay / tuition fee', 2280)]

The synthax 
%%sql 

SELECT `Tuition fees`, COUNT(`Tuition fees`)
FROM university_admission_russia
GROUP BY `Tuition fees`

was used to get the aggregates of the different categories of tuition fees in the data set. 
The tuition fee count was 5016 and the syntax for it is 
%%sql 

SELECT COUNT(`Tuition fees`)
FROM university_admission_russia



In [30]:
%%sql 

SELECT Year, `Branches of Science` as Faculty, SUM(`Number of Students`) as Total_Admission
FROM university_admission_russia
GROUP BY Year, Faculty
ORDER BY Total_Admission DESC

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
100 rows affected.


[(2023, 'Engineering and Technology', Decimal('972960')),
 (2022, 'Engineering and Technology', Decimal('897451')),
 (2020, 'Engineering and Technology', Decimal('855784')),
 (2016, 'Engineering and Technology', Decimal('850676')),
 (2019, 'Engineering and Technology', Decimal('847046')),
 (2018, 'Engineering and Technology', Decimal('832918')),
 (2021, 'Engineering and Technology', Decimal('828855')),
 (2017, 'Engineering and Technology', Decimal('827308')),
 (2023, 'Social Sciences', Decimal('780648')),
 (2022, 'Social Sciences', Decimal('717968')),
 (2021, 'Social Sciences', Decimal('687219')),
 (2020, 'Social Sciences', Decimal('686045')),
 (2019, 'Social Sciences', Decimal('671716')),
 (2017, 'Social Sciences', Decimal('669387')),
 (2018, 'Social Sciences', Decimal('669345')),
 (2016, 'Social Sciences', Decimal('662283')),
 (2015, 'Engineering and Technology', Decimal('455596')),
 (2014, 'Engineering and Technology', Decimal('435116')),
 (2023, 'Healthcare and Medicine', Decimal('220347')),
 (2023, 'Education Sciences and Pedagogy', Decimal('202524')),
 (2022, 'Healthcare and Medicine', Decimal('195369')),
 (2022, 'Education Sciences and Pedagogy', Decimal('194227')),
 (2021, 'Education Sciences and Pedagogy', Decimal('183023')),
 (2021, 'Healthcare and Medicine', Decimal('181634')),
 (2020, 'Healthcare and Medicine', Decimal('176534')),
 (2019, 'Education Sciences and Pedagogy', Decimal('174670')),
 (2018, 'Education Sciences and Pedagogy', Decimal('171660')),
 (2020, 'Education Sciences and Pedagogy', Decimal('170055')),
 (2017, 'Education Sciences and Pedagogy', Decimal('165678')),
 (2019, 'Healthcare and Medicine', Decimal('165230')),
 (2016, 'Education Sciences and Pedagogy', Decimal('163122')),
 (2015, 'Social Sciences', Decimal('162894')),
 (2014, 'Social Sciences', Decimal('159494')),
 (2018, 'Healthcare and Medicine', Decimal('157649')),
 (2017, 'Healthcare and Medicine', Decimal('150980')),
 (2016, 'Healthcare and Medicine', Decimal('146324')),
 (2022, 'Agriculture', Decimal('113875')),
 (2023, 'Agriculture', Decimal('110800')),
 (2020, 'Agriculture', Decimal('107274')),
 (2019, 'Agriculture', Decimal('106936')),
 (2021, 'Agriculture', Decimal('106771')),
 (2018, 'Agriculture', Decimal('105204')),
 (2017, 'Agriculture', Decimal('101562')),
 (2016, 'Agriculture', Decimal('100162')),
 (2015, 'Education Sciences and Pedagogy', Decimal('99829')),
 (2014, 'Education Sciences and Pedagogy', Decimal('96946')),
 (2023, 'Arts and Culture', Decimal('92608')),
 (2023, 'Humanities', Decimal('87312')),
 (2022, 'Arts and Culture', Decimal('86938')),
 (2022, 'Humanities', Decimal('84100')),
 (2021, 'Arts and Culture', Decimal('81118')),
 (2021, 'Humanities', Decimal('79850')),
 (2020, 'Humanities', Decimal('78686')),
 (2019, 'Humanities', Decimal('77382')),
 (2020, 'Arts and Culture', Decimal('76856')),
 (2018, 'Humanities', Decimal('75353')),
 (2015, 'Healthcare and Medicine', Decimal('74060')),
 (2014, 'Healthcare and Medicine', Decimal('73922')),
 (2019, 'Arts and Culture', Decimal('71231')),
 (2017, 'Humanities', Decimal('70983')),
 (2018, 'Arts and Culture', Decimal('69438')),
 (2016, 'Humanities', Decimal('69240')),
 (2017, 'Arts and Culture', Decimal('66846')),
 (2016, 'Arts and Culture', Decimal('65562')),
 (2015, 'Agriculture', Decimal('64049')),
 (2014, 'Agriculture', Decimal('62505')),
 (2023, 'Mathematical and Natural Sciences', Decimal('61568')),
 (2022, 'Mathematical and Natural Sciences', Decimal('61475')),
 (2021, 'Mathematical and Natural Sciences', Decimal('56383')),
 (2019, 'Mathematical and Natural Sciences', Decimal('55295')),
 (2018, 'Mathematical and Natural Sciences', Decimal('55289')),
 (2020, 'Mathematical and Natural Sciences', Decimal('54051')),
 (2017, 'Mathematical and Natural Sciences', Decimal('53066')),
 (2016, 'Mathematical and Natural Sciences', Decimal('51821')),
 (2015, 'Mathematical and Natural Sciences', Decimal('43815')),
 (2015, 'Arts and Culture', Decimal('41053')),
 (2014, 'Arts and C

In [33]:
%%sql 

SELECT Year, SUM(`Number of Applications`) as Total_No_Application, SUM(`Number of Students`) as Total_Admission
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application, Total_Admission 

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, Decimal('5909095'), Decimal('943316')),
 (2015, Decimal('5946155'), Decimal('978366')),
 (2016, Decimal('6591943'), Decimal('2109200')),
 (2017, Decimal('6734044'), Decimal('2105822')),
 (2018, Decimal('7313275'), Decimal('2138046')),
 (2019, Decimal('7689487'), Decimal('2171518')),
 (2020, Decimal('9004885'), Decimal('2207411')),
 (2021, Decimal('9699511'), Decimal('2206934')),
 (2022, Decimal('10682268'), Decimal('2353745')),
 (2023, Decimal('11671419'), Decimal('2531088'))]

In [36]:
%%sql 

SELECT Year, SUM(`Number of Applications`) as Total_No_Application, SUM(`Number of Students`) as Total_Admission, (SUM(`Number of Applications`) -  SUM(`Number of Students`)) as UnSuccessful_Applicatns
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application, Total_Admission 

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, Decimal('5909095'), Decimal('943316'), Decimal('4965779')),
 (2015, Decimal('5946155'), Decimal('978366'), Decimal('4967789')),
 (2016, Decimal('6591943'), Decimal('2109200'), Decimal('4482743')),
 (2017, Decimal('6734044'), Decimal('2105822'), Decimal('4628222')),
 (2018, Decimal('7313275'), Decimal('2138046'), Decimal('5175229')),
 (2019, Decimal('7689487'), Decimal('2171518'), Decimal('5517969')),
 (2020, Decimal('9004885'), Decimal('2207411'), Decimal('6797474')),
 (2021, Decimal('9699511'), Decimal('2206934'), Decimal('7492577')),
 (2022, Decimal('10682268'), Decimal('2353745'), Decimal('8328523')),
 (2023, Decimal('11671419'), Decimal('2531088'), Decimal('9140331'))]

In [50]:
%%sql 

SELECT 
    Year, 
    SUM(`Number of Applications`) as Total_No_Application, 
    SUM(`Number of Students`) as Total_Admission, 
    (SUM(`Number of Applications`) -  SUM(`Number of Students`)) as UnSuccessful_Applicatns,
    ((SUM(`Number of Students`)/SUM(`Number of Applications`))*100) as percentage_Acceptance_Rate,
    ((SUM(`Number of Applications`) -  SUM(`Number of Students`))/SUM(`Number of Applications`))*100) as Percentage_Rej_Rate,
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application, Total_Admission 

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, Decimal('5909095'), Decimal('943316'), Decimal('4965779'), Decimal('15.9638')),
 (2015, Decimal('5946155'), Decimal('978366'), Decimal('4967789'), Decimal('16.4538')),
 (2016, Decimal('6591943'), Decimal('2109200'), Decimal('4482743'), Decimal('31.9966')),
 (2017, Decimal('6734044'), Decimal('2105822'), Decimal('4628222'), Decimal('31.2713')),
 (2018, Decimal('7313275'), Decimal('2138046'), Decimal('5175229'), Decimal('29.2351')),
 (2019, Decimal('7689487'), Decimal('2171518'), Decimal('5517969'), Decimal('28.2401')),
 (2020, Decimal('9004885'), Decimal('2207411'), Decimal('6797474'), Decimal('24.5135')),
 (2021, Decimal('9699511'), Decimal('2206934'), Decimal('7492577'), Decimal('22.7530')),
 (2022, Decimal('10682268'), Decimal('2353745'), Decimal('8328523'), Decimal('22.0341')),
 (2023, Decimal('11671419'), Decimal('2531088'), Decimal('9140331'), Decimal('21.6862'))]

In [54]:
%%sql 

SELECT 
    Year, 
    SUM(`Number of Applications`) as Total_No_Application, 
    SUM(`Number of Students`) as Total_Admission, 
    (SUM(`Number of Applications`) -  SUM(`Number of Students`)) as UnSuccessful_Applicatns,
    ((SUM(`Number of Students`)/SUM(`Number of Applications`))*100) as Percent_Accept_Rate,
    (((SUM(`Number of Applications`) -  SUM(`Number of Students`))/SUM(`Number of Applications`))*100) as Percent_Rej_Rate
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application, Total_Admission

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, Decimal('5909095'), Decimal('943316'), Decimal('4965779'), Decimal('15.9638'), Decimal('84.0362')),
 (2015, Decimal('5946155'), Decimal('978366'), Decimal('4967789'), Decimal('16.4538'), Decimal('83.5462')),
 (2016, Decimal('6591943'), Decimal('2109200'), Decimal('4482743'), Decimal('31.9966'), Decimal('68.0034')),
 (2017, Decimal('6734044'), Decimal('2105822'), Decimal('4628222'), Decimal('31.2713'), Decimal('68.7287')),
 (2018, Decimal('7313275'), Decimal('2138046'), Decimal('5175229'), Decimal('29.2351'), Decimal('70.7649')),
 (2019, Decimal('7689487'), Decimal('2171518'), Decimal('5517969'), Decimal('28.2401'), Decimal('71.7599')),
 (2020, Decimal('9004885'), Decimal('2207411'), Decimal('6797474'), Decimal('24.5135'), Decimal('75.4865')),
 (2021, Decimal('9699511'), Decimal('2206934'), Decimal('7492577'), Decimal('22.7530'), Decimal('77.2470')),
 (2022, Decimal('10682268'), Decimal('2353745'), Decimal('8328523'), Decimal('22.0341'), Decimal('77.9659')),
 (2023, Decimal('11671419'), Decimal('2531088'), Decimal('9140331'), Decimal('21.6862'), Decimal('78.3138'))]

The above is a tabular distri. of the Yr, No. of Appl., Admissions, Percentage Acceptance Rate, and the Percentage Rejection Rate in the Uni for 10 years (2014 to 2023)

N.B: Alias was used in the analysis to help represent the new headings i a prefered manner.
%%sql 

SELECT 
    Year, 
    SUM(`Number of Applications`) as Total_No_Application, 
    SUM(`Number of Students`) as Total_Admission, 
    (SUM(`Number of Applications`) -  SUM(`Number of Students`)) as UnSuccessful_Applicatns,
    ((SUM(`Number of Students`)/SUM(`Number of Applications`))*100) as Percent_Accept_Rate,
    (((SUM(`Number of Applications`) -  SUM(`Number of Students`))/SUM(`Number of Applications`))*100) as Percent_Rej_Rate
FROM university_admission_russia
GROUP BY Year
ORDER BY Total_No_Application, Total_Admission

In [10]:
%%sql 

SELECT *

FROM university_admission_russia
LIMIT 3

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
3 rows affected.


[(1, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Mathematics and Mechanics', 2014, 47923, 7144),
 (2, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Computer Science', 2014, 28787, 3238),
 (3, 'Higher Education', "Bachelor's degree", 'Scholarship / tuition-free', 1, 'Mathematical and Natural Sciences', 'Physics and Astronomy', 2014, 23906, 4391)]

In [11]:
%%sql 

SELECT DISTINCT `Group of Professions`
FROM university_admission_russia
WHERE `Group of Professions` = `Computer Science`

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
57 rows affected.


[('Mathematics and Mechanics',),
 ('Computer Science',),
 ('Physics and Astronomy',),
 ('Chemistry',),
 ('Earth Science',),
 ('Biological Sciences',),
 ('Architecture',),
 ('Construction Equipment and Technologies',),
 ('Information Technology',),
 ('Information Security',),
 ('Electronics, and Radio Engineering and Communication Systems',),
 ('Photonics, Instrument Engineering, Optical and Biotechnical Technologies',),
 ('Electrical and Thermal Power Engineering',),
 ('Nuclear Energy and Technology',),
 ('Mechanical Engineering',),
 ('Physics and Technical Sciences and Technologies',),
 ('Weaponry and Military Technology',),
 ('Chemical Technologies',),
 ('Industrial Ecology and Biotechnology',),
 ('Industrial Safety and Environmental Management',),
 ('Geology, Mining, Oil and Gas Engineering and Geodesy',),
 ('Material Technology',),
 ('Transport Equipment and Technologies',),
 ('Aviation, Rocket, and Space Technology',),
 ('Air Navigation and Operation of Aviation, Rocket and Space Technology',),
 ('Shipbuilding and Water Transport Technologies',),
 ('Technical Management',),
 ('Nanotechnology',),
 ('Light Industry Technologies',),
 ('Fundamental Medicine',),
 ('Clinical Medicine',),
 ('Health Sciences and Preventive Healthcare',),
 ('Pharmacy',),
 ('Nursing',),
 ('Agriculture, Forestry, and Fisheries',),
 ('Veterinary Medicine and Zootechnics',),
 ('Psychology',),
 ('Economics and Management',),
 ('Sociology and Social Services',),
 ('Jurisprudence',),
 ('Political Science and Regional Studies',),
 ('Mass Media and Information',),
 ('Service and Tourism',),
 ('Education Sciences and Pedagogy',),
 ('Linguistics and Philology',),
 ('History and Archaeology',),
 ('Philosophy, Ethics, and Religious Studies',),
 ('Theology',),
 ('Physical Education and Sports',),
 ('Art History',),
 ('Cultural Studies and Socio-Cultural Projects',),
 ('Performing Arts and Literary Creativity',),
 ('Musical Art',),
 ('Design and Restoration',),
 ('Film and Screen Arts',),
 ('Military Management',),
 ('Oriental and African Studies',)]

In [18]:
%%sql 

SELECT `Group of Professions`, SUM(`Number of Students`)
FROM university_admission_russia
WHERE `Group of Professions` = 'Computer Science'
LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
1 rows affected.


[('Computer Science', Decimal('61447'))]

The total number of students offering Computer Science in the Uni.

In [19]:
%%sql 

SELECT `Group of Professions`, SUM(`Number of Applications`)
FROM university_admission_russia
WHERE `Group of Professions` = 'Computer Science'
LIMIT 5

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
1 rows affected.


[('Computer Science', Decimal('574041'))]

The total number of students application for Computer Science in the Uni.

In [5]:
%%sql 

SELECT Year, `Group of Professions`, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY `Group of Professions`  = 'Computer Science'

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
(pymysql.err.OperationalError) (1055, "Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'uni_admission.university_admission_russia.Year' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by")
[SQL: SELECT Year, `Group of Professions`, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY `Group of Professions`]
(Background on this error at: http://sqlalche.me/e/14/e3q8)


In [7]:
%%sql 

SELECT `Group of Professions`, Year, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY  Year,`Group of Professions`= 'Computer Science`

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
570 rows affected.


[('Mathematics and Mechanics', 2014, Decimal('9790')),
 ('Computer Science', 2014, Decimal('4012')),
 ('Physics and Astronomy', 2014, Decimal('6506')),
 ('Chemistry', 2014, Decimal('4269')),
 ('Earth Science', 2014, Decimal('9128')),
 ('Biological Sciences', 2014, Decimal('5833')),
 ('Architecture', 2014, Decimal('4965')),
 ('Construction Equipment and Technologies', 2014, Decimal('54705')),
 ('Information Technology', 2014, Decimal('54245')),
 ('Information Security', 2014, Decimal('5774')),
 ('Electronics, and Radio Engineering and Communication Systems', 2014, Decimal('21192')),
 ('Photonics, Instrument Engineering, Optical and Biotechnical Technologies', 2014, Decimal('5776')),
 ('Electrical and Thermal Power Engineering', 2014, Decimal('36580')),
 ('Nuclear Energy and Technology', 2014, Decimal('2181')),
 ('Mechanical Engineering', 2014, Decimal('43581')),
 ('Physics and Technical Sciences and Technologies', 2014, Decimal('1479')),
 ('Weaponry and Military Technology', 2014, Decimal('675')),
 ('Chemical Technologies', 2014, Decimal('12396')),
 ('Industrial Ecology and Biotechnology', 2014, Decimal('28012')),
 ('Industrial Safety and Environmental Management', 2014, Decimal('12767')),
 ('Geology, Mining, Oil and Gas Engineering and Geodesy', 2014, Decimal('29921')),
 ('Material Technology', 2014, Decimal('13018')),
 ('Transport Equipment and Technologies', 2014, Decimal('65776')),
 ('Aviation, Rocket, and Space Technology', 2014, Decimal('5296')),
 ('Air Navigation and Operation of Aviation, Rocket and Space Technology', 2014, Decimal('4192')),
 ('Shipbuilding and Water Transport Technologies', 2014, Decimal('12182')),
 ('Technical Management', 2014, Decimal('12432')),
 ('Nanotechnology', 2014, Decimal('827')),
 ('Light Industry Technologies', 2014, Decimal('7144')),
 ('Fundamental Medicine', 2014, Decimal('445')),
 ('Clinical Medicine', 2014, Decimal('38292')),
 ('Health Sciences and Preventive Healthcare', 2014, Decimal('1606')),
 ('Pharmacy', 2014, Decimal('4014')),
 ('Nursing', 2014, Decimal('29565')),
 ('Agriculture, Forestry, and Fisheries', 2014, Decimal('49265')),
 ('Veterinary Medicine and Zootechnics', 2014, Decimal('13240')),
 ('Psychology', 2014, Decimal('4952')),
 ('Economics and Management', 2014, Decimal('88918')),
 ('Sociology and Social Services', 2014, Decimal('10175')),
 ('Jurisprudence', 2014, Decimal('22314')),
 ('Political Science and Regional Studies', 2014, Decimal('3795')),
 ('Mass Media and Information', 2014, Decimal('5427')),
 ('Service and Tourism', 2014, Decimal('23913')),
 ('Education Sciences and Pedagogy', 2014, Decimal('96946')),
 ('Linguistics and Philology', 2014, Decimal('9683')),
 ('History and Archaeology', 2014, Decimal('7686')),
 ('Philosophy, Ethics, and Religious Studies', 2014, Decimal('1106')),
 ('Theology', 2014, Decimal('287')),
 ('Physical Education and Sports', 2014, Decimal('16505')),
 ('Art History', 2014, Decimal('573')),
 ('Cultural Studies and Socio-Cultural Projects', 2014, Decimal('11457')),
 ('Performing Arts and Literary Creativity', 2014, Decimal('2730')),
 ('Musical Art', 2014, Decimal('12873')),
 ('Design and Restoration', 2014, Decimal('12028')),
 ('Film and Screen Arts', 2014, Decimal('866')),
 ('Military Management', 2014, Decimal('1')),
 ('Oriental and African Studies', 2014, Decimal('0')),
 ('Mathematics and Mechanics', 2015, Decimal('11195')),
 ('Computer Science', 2015, Decimal('4399')),
 ('Physics and Astronomy', 2015, Decimal('7362')),
 ('Chemistry', 2015, Decimal('4465')),
 ('Earth Science', 2015, Decimal('9999')),
 ('Biological Sciences', 2015, Decimal('6395')),
 ('Architecture', 2015, Decimal('5325')),
 ('Construction Equipment and Technologies', 2015, Decimal('55177')),
 ('Information Technology', 2015, Decimal('58252')),
 ('Information Security', 2015, Decimal('6368')),
 ('Electronics, and Radio Engineering and Communication Systems', 2015, Decimal('22760')),
 ('Photonics, Instrument Engineering, Optical and Biotechnical Technologies', 2015, Decimal

In [12]:
%%sql 

SELECT `Group of Professions`, Year, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY Year, `Group of Professions`
HAVING `Group of Professions`= 'Computer Science`

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
(pymysql.err.ProgrammingError) (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''Computer Science`' at line 4")
[SQL: SELECT `Group of Professions`, Year, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY Year, `Group of Professions`
HAVING `Group of Professions`= 'Computer Science`]
(Background on this error at: http://sqlalche.me/e/14/f405)


In [17]:
%%sql 
SELECT Year, `Group of Professions`, SUM(`Number of Students`)
FROM university_admission_russia
GROUP BY Year, `Group of Professions`
HAVING `Group of Professions` = 'Computer Science';

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, 'Computer Science', Decimal('4012')),
 (2015, 'Computer Science', Decimal('4399')),
 (2016, 'Computer Science', Decimal('5472')),
 (2017, 'Computer Science', Decimal('5879')),
 (2018, 'Computer Science', Decimal('5939')),
 (2019, 'Computer Science', Decimal('6354')),
 (2020, 'Computer Science', Decimal('6397')),
 (2021, 'Computer Science', Decimal('6940')),
 (2022, 'Computer Science', Decimal('7760')),
 (2023, 'Computer Science', Decimal('8295'))]

In [19]:
%%sql 
SELECT Year, `Group of Professions` as Department, SUM(`Number of Students`) as Total_No_students
FROM university_admission_russia
GROUP BY Year, `Group of Professions`
HAVING `Group of Professions` = 'Computer Science';

 * mysql+pymysql://root:***@127.0.0.1:3306/uni_admission
10 rows affected.


[(2014, 'Computer Science', Decimal('4012')),
 (2015, 'Computer Science', Decimal('4399')),
 (2016, 'Computer Science', Decimal('5472')),
 (2017, 'Computer Science', Decimal('5879')),
 (2018, 'Computer Science', Decimal('5939')),
 (2019, 'Computer Science', Decimal('6354')),
 (2020, 'Computer Science', Decimal('6397')),
 (2021, 'Computer Science', Decimal('6940')),
 (2022, 'Computer Science', Decimal('7760')),
 (2023, 'Computer Science', Decimal('8295'))]